src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -189,45 +175,27 @@ else if (module instanceof TreeWalkerFilter) {
     @Override
     protected void processFiltered(File file, FileText fileText) throws CheckstyleException {
         // check if already checked and passed the file
-        if (CommonUtils.matchesFileExtension(file, getFileExtensions())) {
-            final String msg = "%s occurred during the analysis of file %s.";
-            final String fileName = file.getPath();
-
-            try {
-                if (!ordinaryChecks.isEmpty()
-                        || !commentChecks.isEmpty()) {
-                    final FileContents contents = new FileContents(fileText);
-                    final DetailAST rootAST = parse(contents);
-
-                    if (!ordinaryChecks.isEmpty()) {
-                        walk(rootAST, contents, AstState.ORDINARY);
-                    }
-                    if (!commentChecks.isEmpty()) {
-                        final DetailAST astWithComments = appendHiddenCommentNodes(rootAST);
-
-                        walk(astWithComments, contents, AstState.WITH_COMMENTS);
-                    }
-                    if (filters.isEmpty()) {
-                        addMessages(messages);
-                    }
-                    else {
-                        final SortedSet<LocalizedMessage> filteredMessages =
-                            getFilteredMessages(fileName, contents, rootAST);
-                        addMessages(filteredMessages);
-                    }
-                    messages.clear();
-                }
+        if (CommonUtils.matchesFileExtension(file, getFileExtensions())
+                && (!ordinaryChecks.isEmpty() || !commentChecks.isEmpty())) {
+            final FileContents contents = new FileContents(fileText);
+            final DetailAST rootAST = JavaParser.parse(contents);
+
+            if (!ordinaryChecks.isEmpty()) {
+                walk(rootAST, contents, AstState.ORDINARY);
             }
-            catch (final TokenStreamRecognitionException tre) {
-                final String exceptionMsg = String.format(Locale.ROOT, msg,
-                        "TokenStreamRecognitionException", fileName);
-                throw new CheckstyleException(exceptionMsg, tre);
+            if (!commentChecks.isEmpty()) {
+                final DetailAST astWithComments = JavaParser.appendHiddenCommentNodes(rootAST);
+                walk(astWithComments, contents, AstState.WITH_COMMENTS);
             }
-            catch (RecognitionException | TokenStreamException ex) {
-                final String exceptionMsg = String.format(Locale.ROOT, msg,
-                        ex.getClass().getSimpleName(), fileName);
-                throw new CheckstyleException(exceptionMsg, ex);
```

Comment:
I question the removal of this try/catch. What if exception is somewhere else in TreeWalker besides parsing and we need to add more catch clauses here. Also now that try/catch is in Parser, everything that calls it will override original exception like `Main` and `AstTreeStringPrinter` which are only working with 1 file anyways.
I think we should restore this try/catch here and remove it from parser.

Reply:
The try/catch weren't removed. They were _moved_ to the `JavaParser.parse()` method:
https://github.com/Softus/checkstyle/blob/e12f5e8b7c5f5970c242a77e0d51877c0f5db183/src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java#L98

Reply:
> weren't removed

@pbludov I meant they were removed from `TreeWalker` which is what I am questioning.
As I stated, I think this should stay in `TreeWalker`. All other classes calling `JavaParser` is only dealing with a single file anyways.

Reply:
@rnveach, in this case `JavaParser.parseFile(File, options)` will throw too many exceptions.
Now:
```
    /**
     * Parses Java source file.
     * @param file the file to parse
     * @param options {@link Options} to control inclusion of comment nodes
     * @return DetailAST tree
     * @throws IOException if the file could not be read
     * @throws CheckstyleException if the file is not a valid Java source file
     */
    public static DetailAST parseFile(File file, Options options)
            throws IOException, CheckstyleException {
        final FileText text = new FileText(file.getAbsoluteFile(),
            System.getProperty("file.encoding", StandardCharsets.UTF_8.name()));
        return parse(new FileContents(text), options);
    }
```
You want
```
    /**
     * Parses Java source file.
     * @param file the file to parse
     * @param options {@link Options} to control inclusion of comment nodes
     * @return DetailAST tree
     * @throws IOException if the file could not be read
     * @throws TokenStreamException if lexing failed
     * @throws RecognitionException if parsing failed
     */
    public static DetailAST parseFile(File file, Options options)
            throws IOException, TokenStreamException, RecognitionException {
        final FileText text = new FileText(file.getAbsoluteFile(),
            System.getProperty("file.encoding", StandardCharsets.UTF_8.name()));
        return parse(new FileContents(text), options);
    }
```
Only two exceptions per method are allowed for the CS code.
Also, this will expose some ANTLR internals out of the JavaParser. Many classes will need to import classes from the ANTLR package. This will create some extra dependencies for no good reason.

Reply:
What if we turn antlr exceptions into checkstyle exceptions in parser, and change TreeWalker catch to catch checkstyle exception.

@romani What do you think of this change or are you fine with the way it is coded now?

Reply:
>What if we turn antlr exceptions into checkstyle exceptions in parser, and change TreeWalker catch to catch checkstyle exception.

It's possible. I will do it this way. But I still do not understand the reason.

Reply:
done.

Reply:
@pbludov , @rnveach ,

"RecognitionException | TokenStreamException" are both Checked exceptions so, if we move code to Parser, catches should be moved too, as they can not happen in Treewalker.

I do not see a reason in ParserException (especially in API, and you wrap it (do not let it go to users code) .... blow my mind, sorry). CheckstyleException should be enough.

> Only two exceptions per method are allowed for the CS code.

in general it is true, I do not see good reason to make special suppression here, but if can prove we can do suppress for this case.